### ISSUE TITLE
Improve commit display

### ISSUE THREAD

**Issue by SoftCreatR** [gogits/gogs#2062]
If many files / lines have been changed within a commit, gogs just shows "[W] Diff data too large" which is fine. However, in this case, it doesn't show anything else, which is annoying, because in this case, you'll have to use your external git client to check out the changes.

I would suggest, that if the diff data is too large, gogs omits the diff but still shows the files that have been changed in the commit.

**Comment by Unknwon** [comment]
Thanks your feedback!

> I would suggest, that if the diff data is too large, gogs omits the diff but still shows the files that have been changed in the commit.

What is the benefit of this?

**Comment by SoftCreatR** [comment]
> What is the benefit of this?

That people still can see, what's changed. Currently, they see nothing.

**Comment by Unknwon** [comment]
I think what you need is this: https://gogs.io/docs/advanced/configuration_cheat_sheet#git-(git)

**Comment by SoftCreatR** [comment]
No - https://github.com/gogits/gogs/issues/2051

I cannot always Increase the setting just to get the list of changed files. So what i suggest is, that if the number of changed lines exceeds `MAX_GIT_DIFF_LINES` it should still show the changed files.

**Comment by Unknwon** [comment]
OK, I got it. Thanks for clarifying.

**Comment by xxxtonixxx** [comment]
+1, and would it be possible that files with too lines were skipped?

**Comment by Unknwon** [comment]
` too lines`?

**Comment by xxxtonixxx** [comment]
Lines of code that they have changed or they have been added

Sorry! :sweat_smile:

**Comment by Unknwon** [comment]
OK, yeah, make sense!

**Comment by Ancyker** [comment]
+1 I'd also like to see this

**Comment by SoftCreatR** [comment]
Any update here?

Btw... This PR seems to do, what i expect: https://github.com/gogits/gogs/pull/3174

**Comment by andreynering** [comment]
@SoftCreatR Yes, I did the PR for better handling this case (https://github.com/gogits/gogs/pull/3174).

It's already working, just waiting for merge. You could build from source if you don't want to wait.

**Comment by jspivaktq** [comment]
Another suggestion ( valuable for me) would be to display a **partial** list of changes in the usual manner with a conspicuous warning that it is not complete.

the benefit of this is that some changes (i just made 30k+ to one file) are repetitive.  For example changing a variable name or spelling.  being able to see - even a partial view of - changes would give me confidence that the change i was about to commit was what i intended.

**Comment by Unknwon** [comment]
>Another suggestion ( valuable for me) would be to display a partial list of changes in the usual manner with a conspicuous warning that it is not complete.

This is what Gogs does currently.

**Comment by tetrap** [comment]
+1
looks like very basic feature to have
